internal/models/sessions.go

```diff
@@ -290,40 +292,22 @@ func (s *Session) CalculateAALAndAMR(tx *storage.Connection) (aal string, amr []
 	}
 
 	// makes sure that the AMR claims are always ordered most-recent first
-
-	// sort in ascending order
-	sort.Sort(sortAMREntries{
-		Array: amr,
-	})
-
-	// now reverse for descending order
-	_ = sort.Reverse(sortAMREntries{
+	sort.Sort(sort.Reverse(sortAMREntries{
 		Array: amr,
-	})
-
-	lastIndex := len(amr) - 1
```

Comment:
This is a handful. Here are the changes in the section:

1. Chain the sort methods
2. Start from 0 instead of from 1
3. Use the `IsForSSOProvider` method on identities
4.  Reduce nesting by extracting conditions